sources/lib/frontend/cnf.ml

```diff
@@ -34,6 +34,49 @@ open Commands
 module E = Expr
 module Sy = Symbols
 module SE = E.Set
+module SRE = Simple_reasoner_expr
+
+
+
+let choose_preproc () :
+  (module Simple_reasoner_expr.S with type expr = Expr.t
+                                  and type env = Theory.Main_Default.t
+                                  and type expl = Explanation.t)  =
+  if Options.simplify_th ()
```

Comment:
Isn't this if-then-else inversed ? The way I read it: if options specify to use theory simplification, then the module returned has a dummy query implementation, else it returns the actual query function sued for simplification...

Also, since only the implementation of the query function changes depending on the options, wouldn't it be simpler to simply check that option in the query function, and avoid the dependant instanciation of the functor ?